x-pack/plugins/fleet/server/services/download_source.ts

```diff
@@ -12,15 +12,15 @@ import {
   DEFAULT_DOWNLOAD_SOURCE_ID,
 } from '../constants';
 
-import type { DownloadSource, DownloadSourceAttributes, DownloadSourceBase } from '../types';
+import type { DownloadSource, DownloadSourceSOAttributes, DownloadSourceBase } from '../types';
 import { DownloadSourceError, FleetError } from '../errors';
 import { SO_SEARCH_LIMIT } from '../../common';
 
 import { agentPolicyService } from './agent_policy';
 import { appContextService } from './app_context';
 import { escapeSearchQueryPhrase } from './saved_object';
 
-function savedObjectToDownloadSource(so: SavedObject<DownloadSourceAttributes>) {
+function savedObjectToDownloadSource(so: SavedObject<DownloadSourceSOAttributes>) {
   const { source_id: sourceId, ...attributes } = so.attributes;
```

Comment:
I think we have to refactor mapper functions like this to avoid creating a response object as `...attributes`.
Here we can map directly to known properties of `DownloadSource` type:
```
return {
  id: sourceId ?? so.id,
  name: attributes.name,
  host: attributes.name,
  is_default: attributes.is_default
}
```

Reply:
Sure, I can file another follow up ticket for it